illumos-utils/src/fakes/dladm.rs

```diff
@@ -0,0 +1,47 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::dladm::Api;
+use crate::dladm::CreateVnicError;
+use crate::dladm::DeleteVnicError;
+use crate::dladm::FindPhysicalLinkError;
+use crate::dladm::VnicSource;
+use crate::link::Link;
+use omicron_common::api::external::MacAddr;
+use omicron_common::vlan::VlanID;
+use std::sync::Arc;
+
+/// A fake implementation of [crate::dladm::Dladm].
+///
+/// This struct implements the [crate::dladm::Api] interface but avoids
```

Comment:
(This comment also applies to `zone.rs`) this could be more real, but this was the "minimal API required" to replace most mock-based tests.